integration/airflow/openlineage/airflow/extractors/sagemaker_extractors.py

```diff
@@ -0,0 +1,272 @@
+        artifact_summaries = sagemaker_client.list_artifacts(SourceUri=model_package_arn)['ArtifactSummaries']
+
+        model_building_datasets = []
+        model_artifact_arns = []
+
+        for summary in artifact_summaries:
+            if summary['ArtifactType'] == 'Model':
+                model_artifact_arns.append(summary['ArtifactArn'])
+
+        for model_artifact_arn in model_artifact_arns:
+            try:
+                response = sagemaker_client.query_lineage(
+                    StartArns=[
+                        model_artifact_arn,
+                    ],
+                    Direction='Ascendants',
+                    IncludeEdges=False,
+                    Filters={
+                        'Types': [
+                            'DataSet',
+                            'TrainingJob'
+                        ],
+                        'LineageTypes': [
+                            'TrialComponent',
+                            'Artifact'
+                        ],
+                    },
+                    MaxDepth=10,
+                    MaxResults=10,
+                )
+
+                for vertex in response['Vertices']:
+                    if vertex['Type'] == 'DataSet':
+                        model_building_datasets.append(
+                            sagemaker_client.describe_artifact(ArtifactArn=vertex['Arn'])['Source']['SourceUri']
+                        )
+
+                    if vertex['Type'] == 'TrainingJob':
+                        response = sagemaker_client.describe_trial_component(
+                            TrialComponentName=vertex['Arn'].split('/')[-1]
+                        )
+                        training_job_name = response['Source']['SourceArn'].split('/')[-1]
+
+                        training_job = sagemaker_client.describe_training_job(TrainingJobName=training_job_name)
+
+                        start_time = training_job['TrainingStartTime']
+                        end_time = training_job['TrainingEndTime']
+                        training_job_arn = training_job['TrainingJobArn']
+                        hyper_parameters = training_job['HyperParameters']
+                        inputs = [i['DataSource']['S3DataSource']['S3Uri'] for i in training_job['InputDataConfig']]
+                        output = training_job['ModelArtifacts']['S3ModelArtifacts']
+
+                        self._create_sagemaker_training_job(training_job_name, inputs, output, hyper_parameters,
+                                                            start_time, end_time, training_job_arn)
+            except Exception as e:
+                log.error(f"Error getting/creating sagemaker model lineage data. {e}", exc_info=True)
+
+    def _create_sagemaker_training_job(self, training_job_name, data_inputs, model_output, hyper_parameters, start_time,
+                                       end_time, job_arn):
+
+        client = OpenLineageClient.from_environment()
```

Comment:
You don't need to explicitly create OpenLineage Client and emit events by yourself. It is enough to return `TaskMetadata` from `extract` or `extract_on_complete` method.

Reply:
This is to create an Openlineage job that represents a sagemaker traning job. That training job is not executed as part of any dag so we need it to show lineage of training data inputs and model outputs. 

I may change this approach to creating an extractor for the `SageMakerTrainingOperator` soon, and then this may not be as important.

Reply:
@JDarDagran I've gone ahead and removed this and added the `SageMakerTrainingOperator` in its place.